### PR TITLE
PLAT-104265: Fix comments for Documetation

### PR DESCRIPTION
**Pull request by jeongran1230**
Enact-DCO-1.0-Signed-off-by: Jeongran Jang <jeongran.jang@lge.com>

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Some example code and tags are not working properly in enact docs.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Fix comments for AgateDecorator, Arc, Icon, Picker, Scroller, Spinner and Drawser

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-104265

### Comments

### CODE REVIEW

File: Drawer/Drawer.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * An Agate-themed drawer component.
  *
  * @example
- * <Drawer header="Surprise">Hello Enact!</Drawer>
+ * <Drawer style={{marginTop:'60px'}} open>Hello Enact!</Drawer>
```
Comment (jeonghee27) [inline Drawer/Drawer.js:5]: I think height it also needed to prevent cut off.
    margin-top: 33px;
    height: 100px;

File: AgateDecorator/AgateDecorator.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /**
- * Applies Agate specific styling and behaviors.
+ * {@link agate/AgateDecorator} will be removed in 2.0.0. Use {@link agate/ThemeDecorator} instead.
  *
  * @module agate/AgateDecorator
  * @exports AgateDecorator
- * @deprecated Will be removed in 2.0.0. Use {@link agate/ThemeDecorator} instead.
+ * @deprecated
```
Comment (jeonghee27) [inline AgateDecorator/AgateDecorator.js:6]: I'm not should this part it right.  @ybsung  Please check when you merge it.

File: Drawer/Drawer.js [hunk 3]
```diff
@@ -28,6 +28,7 @@ import componentCss from './Drawer.module.less';
 /**
  * A drawer component, without behaviors.
  *
+ *
```
Comment (jeonghee27) [inline Drawer/Drawer.js:31]: Remove empty line

File: Icon/Icon.js [hunk 4]
```diff
@@ -2,7 +2,7 @@
  * Provides Agate styled icon components and behaviors.
  *
  * @example
- * <Icon>flag</Icon>
+ * <Icon>closex</Icon>
```
Comment (jeonghee27) [inline Icon/Icon.js:5]: Why did you change it?

File: Picker/Picker.js [hunk 5]
```diff
@@ -147,7 +147,6 @@ const PickerBase = kind({
  * @class Picker
  * @memberof agate/Picker
  * @extends agate/Picker.PickerBase
- * @mixes agate/internal/Picker.PickerDecorator
```
Comment (jeonghee27) [inline Picker/Picker.js:150]: Why did you remove this part?

File: AgateDecorator/AgateDecorator.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
 /**
- * Applies Agate specific styling and behaviors.
+ * {@link agate/AgateDecorator} will be removed in 2.0.0. Use {@link agate/ThemeDecorator} instead.
```
Comment (jeonghee27) [inline AgateDecorator/AgateDecorator.js:2]: There are another broken links. (not this Component)

agate/ColorPicker
agate/ContextualPopupDecorator/
agate/Input
agate/Panels
....

Please search "@see" in the agate

**Comment by jeonghee27** [comment]
There is no example SwitchItem / ImageItem, RadioItem

**Comment by jeonghee27** [comment]
Could you check the exampler for DatePicker/ TimePicker?? it looks weird

**Comment by jeonghee27** [comment]
When I click a button in SliderButton example, there is a console error.
It could be a Component error, If you cannot handle it in the PR, Please leave the comment on the next ticket to prevent forgetting.

**Comment by jeonghee27** [comment]
I think `PopupMenu` also need margin-top likes `Drawer`

File: Drawer/Drawer.js (hunk 1 above)
Comment (ybsung) [reply Drawer/Drawer.js:5]: Need to use not px but ri.pixelToRem

File: Icon/Icon.js (hunk 4 above)
Comment (jeongran1230) [reply Icon/Icon.js:5]: `flag` is not supported in agate item.

File: Picker/Picker.js (hunk 5 above)
Comment (jeongran1230) [reply Picker/Picker.js:150]: I think that all components included in internal directory  look private components, so line 150 should be removed but I'll recover the code because it is out of the scope in this ticket.

File: AgateDecorator/AgateDecorator.js (hunk 6 above)
Comment (jeongran1230) [reply AgateDecorator/AgateDecorator.js:2]: That's good point, I found the probluem It's not only agate but also all other UI frameworks. 
On local server is linking but I cannot see the link on the web server. Currently I don't know what is different between them. So, how about fix them another ticket?

exmaple) ColorPicker `* @see {@link ui/Changeable.Changeable} ` on local
```
<div class="--see">
 ::before
  "See: "
  <span>
    <a data-tooltip="ui/Chanageable" href="/docs/modules/ui/Changeable/@Changeable">ui/Changeable.Changeable</a>
  </span>
</div>
```
whereas on jenkins server
```
<div class="--see">
 ::before
  "See: "
  ::after
</div>
```

**Review by jeonghee27 — APPROVED** [review]
There are many parts that are difficult for me to judge, so I just approve it. 
@ybsung  Please check my unresolved comment when you merge.

**Comment by jeongran1230** [comment]
missed / omitted / wrong comments are out of scope. we will do it another tickets.
> There is no example SwitchItem / ImageItem / RadioItem

Right, It looks weired but it's basic design supported by each component.
>Could you check the example for DatePicker/ TimePicker?? it looks weird 

It need more anaysis to provide an example for floating based compoenent such as Drawer and PopupMenu. I don't want to deal with them here.
> I think PopupMenu also need margin-top likes Drawer

File: AgateDecorator/AgateDecorator.js (hunk 2 above)
Comment (ybsung) [reply AgateDecorator/AgateDecorator.js:6]: Could you share why did you change the JS Doc here for the deprecated?

File: Scroller/Scroller.js [hunk 7]
```diff
@@ -3,7 +3,7 @@
  *
  * @example
  * <Scroller>
- * 	<div style={{height: '300px'}}>
+ * 	<div style={{height: '100px'}}>
```
Comment (ybsung) [inline Scroller/Scroller.js:6]: Please use ri.scaleToRem instead of px.

**Comment by ybsung** [comment]
image
The ColorPicker sample in Docs is clipped.

**Comment by ybsung** [comment]
How about ` * [Group]{@link ui/Group.Group}` instead of ` * [Group]{@link ui/Group}`?
https://github.com/enactjs/agate/blob/develop/CheckboxItem/CheckboxItem.js#L32

By doing it, when hovering the lint, the tooltip will be shown.
image

**Comment by ybsung** [comment]
image

It looks that the DatePicker sample is broken.

**Comment by ybsung** [comment]
image

Image example code is clipped.

**Comment by ybsung** [comment]
image

The sample above is clipped.

**Comment by ybsung** [comment]
There are no sample in JSDoc for Panels.

**Comment by ybsung** [comment]
image

The sample is not shown.

**Comment by ybsung** [comment]
image

The internal Picker might not be shown in Documentation.

**Comment by ybsung** [comment]
As I commented above, please check
- all sample code are shown properly not being clipped
- all samples are shown properly not being clipped
- all samples are not broken

File: AgateDecorator/AgateDecorator.js (hunk 2 above)
Comment (jeongran1230) [reply AgateDecorator/AgateDecorator.js:6]: That's because there is no reason to @deprecated tag with the sentance there. 
`@deprecated Will be removed in 2.0.0. Use {@link agate/ThemeDecorator} instead.`
' Will be removed in 2.0.0. Use {@link agate/ThemeDecorator} instead.'
Both are same results on the doc page

File: Scroller/Scroller.js (hunk 7 above)
Comment (jeongran1230) [reply Scroller/Scroller.js:6]: Will be fixed to the next patch

**Comment by jeongran1230** [comment]
I don't know exactly rules to represent @link
someone wrote @link ui/touchable.touchable, another wrote just module name. 
Could you explain the difference ?
> How about * [Group]{@link ui/Group.Group} instead of * [Group]{@link ui/Group}?

Clipped frame will be solved at /PLAT-120786 because of wide scope.

DatePicker and switch has no problem. please check them into sample app but I agree they need to be fixed in better way.

(out of scope) I don't think Adding @example is not in this scope but they are definitely added.
> There are no @example or Usage in JSDoc for Panels, RadioItem, TooltipDecorator, VirtualList.

It's shown actually.
> The sample is not shown.

(out of scope) I explained this before, There are so many tags being wrong. I know it should be private.
> The internal Picker might not be shown in Documentation.
